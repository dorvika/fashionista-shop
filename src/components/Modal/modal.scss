.modal {
  box-shadow: 0 15px 15px rgba(0, 0, 0, 0.2);
  background-color: white;
  color: black;
  text-align: center;
  padding: 10px 20px 20px 20px;
  width: 30rem;
  z-index: 10;
  position: fixed;
  top: 20vh;
  left: calc(50% - 15rem);
  display: none;
  &.active {
    display: block;
  }

  &__header {
    display: flex;
    justify-content: flex-end;
    align-items: center;

    &__btn {
      font-size: 28px;
      font-weight: 200;
      border: none;
      background-color: transparent;
      color: black;
      cursor: pointer;
      transition: all 0.4s ease-in;

      &:hover {
        font-weight: 600;
      }
    }
  }

  &__content {
    &__text {
      line-height: 24px;
      padding: 20px 0;
      font-weight: 200;
    }

    &__btn {
      margin: 20px 10px;
      margin-bottom: 0;
      padding: 10px;
      background-color: black;
      color: white;
      border: 1px solid black;
      cursor: pointer;
      font-size: 16px;
      transition: all 0.5s ease-in;
      font-weight: 200;

      &:hover {
        background-color: white;
        color: black;
      }
    }
  }
}

.backdrop {
  position: fixed;
  z-index: 1;
  background-color: rgba(0, 0, 0, 0.7);
  width: 100%;
  height: 100vh;
  top: 0;
  left: 0;
  display: none;
  &.active {
    display: block;
  }
}

@media screen and (max-width: 500px) {
  .modal {
    width: 300px;
    left: calc(50% - 150px);
  }
}
