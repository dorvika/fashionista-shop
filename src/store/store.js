import {
  legacy_createStore as createStore,
  combineReducers,
  compose,
  applyMiddleware,
} from "redux";
import thunk from "redux-thunk";
import productReducer from "./products/reducer";
import favoriteReducer from "./favorities/reducer";
import basketReducer from "./basket/reducer";
import modalReducer from "./modal/reducer";
import { setProductsToLS } from "../helpers/localStorage";

export const rootReducer = combineReducers({
  products: productReducer,
  favorites: favoriteReducer,
  basket: basketReducer,
  modal: modalReducer,
});

const devTools = window.__REDUX_DEVTOOLS_EXTENSION__
  ? window.__REDUX_DEVTOOLS_EXTENSION__()
  : (f) => f;

const syncWithLS = (store) => (next) => (action) => {
  const result = next(action);
  if (action.type === "TOGGLE_FAVORITE") {
    setProductsToLS("Products in Favorite", store.getState().favorites);
    return result;
  }
  if (action.type === "ADD_TO_BASKET" || action.type === "DELETE_FROM_BASKET") {
    setProductsToLS("Products in Basket", store.getState().basket);
    return result;
  }
  return next(action);
};

const store = createStore(
  rootReducer,
  compose(applyMiddleware(thunk, syncWithLS), devTools)
);

export default store;
