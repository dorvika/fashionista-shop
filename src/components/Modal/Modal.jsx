import "./modal.scss";
import { useDispatch, useSelector } from "react-redux";
import { closeModal, modalActionHandler } from "../../store/modal/actions";
import { modal } from "../../store/modal/reducer";
import Button from "../Button";

export default function Modal() {
  const dispatch = useDispatch();
  const { isOpen, content, action, product, deleteTotallyFlag } =
    useSelector(modal);

  const modalHandler = () => {
    dispatch(modalActionHandler(action, product, deleteTotallyFlag));
    onClose();
  };

  const onClose = () => {
    dispatch(closeModal());
  };

  return (
    <>
      <div className={isOpen ? "modal active" : "modal"}>
        <div className="modal__header">
          <Button className="modal__header__btn" onClick={onClose}>
            &times;
          </Button>
        </div>
        <div className="modal__content">
          <p className="modal__content__text">{content}</p>
          <Button className="modal__content__btn" onClick={modalHandler}>
            Ок
          </Button>
          <Button className="modal__content__btn" onClick={onClose}>
            Скасувати
          </Button>
        </div>
      </div>
      <div
        className={isOpen ? "backdrop active" : "backdrop"}
        onClick={onClose}
      />
    </>
  );
}
