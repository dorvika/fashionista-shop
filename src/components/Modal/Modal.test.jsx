import { render, screen } from "@testing-library/react";
import Modal from "./Modal";
import { Provider } from "react-redux";
import { legacy_createStore as createStore } from "redux";
import { rootReducer } from "../../store/store";

describe("<Modal /> component", () => {
  const store = createStore(rootReducer, {
    modal: {
      isOpen: false,
      content: null,
      action: null,
      product: {},
      deleteTotallyFlag: false,
    },
  });

  it("should render modal with three buttons", () => {
    render(
      <Provider store={store}>
        <Modal />
      </Provider>
    );
    screen.getByRole("button", { name: /×/i });
    screen.getByRole("button", { name: /ок/i });
    screen.getByRole("button", { name: /скасувати/i });
  });
  it("should match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <Modal />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
