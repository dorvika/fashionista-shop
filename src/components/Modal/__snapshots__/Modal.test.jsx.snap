// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> component should match snapshot 1`] = `
<div>
  <div
    class="modal"
  >
    <div
      class="modal__header"
    >
      <button
        class="modal__header__btn"
        type="button"
      >
        ×
      </button>
    </div>
    <div
      class="modal__content"
    >
      <p
        class="modal__content__text"
      />
      <button
        class="modal__content__btn"
        type="button"
      >
        Ок
      </button>
      <button
        class="modal__content__btn"
        type="button"
      >
        Скасувати
      </button>
    </div>
  </div>
  <div
    class="backdrop"
  />
</div>
`;

exports[`<Modal /> component should render 1`] = `
<div>
  <div
    class="modal"
  >
    <div
      class="modal__header"
    >
      <button
        class="modal__header__btn"
        type="button"
      >
        ×
      </button>
    </div>
    <div
      class="modal__content"
    >
      <p
        class="modal__content__text"
      />
      <button
        class="modal__content__btn"
        type="button"
      >
        Ок
      </button>
      <button
        class="modal__content__btn"
        type="button"
      >
        Скасувати
      </button>
    </div>
  </div>
  <div
    class="backdrop"
  />
</div>
`;
